File: py/leetcode/P11.py
def maxArea(self, height: List[int]) -> int:
    res = 0
    left = 0
    right = len(height)-1
    while left < right:
        h = min(height[left],height[right])
        res = max(res,h*(right-left))
        if height[left] < height[right]: left+=1
        else: right-=1
    return res

File: src/LeetCode/P11.java
package LeetCode;

/**
 * @desc:
 * @auther: kid1999
 * @date: 2020/4/18 11:05
 **/
public class P11 {
    public static void main(String[] args) {

    }

    public int maxArea(int[] height) {
        int res = 0;
        int l = 0;
        int r =height.length-1;
        while (l<r){
            int h = Math.min(height[l],height[r]);
            res = Math.max(res, h*(r-l));
            if(height[l] < height[r]) l++;
            else r--;
        }
        return res;
    }
}
